referenceframe/worldstate.go

```diff
@@ -1,51 +1,62 @@
 package referenceframe
 
 import (
-	"fmt"
 	"strconv"
 
 	commonpb "go.viam.com/api/common/v1"
 
-	spatial "go.viam.com/rdk/spatialmath"
+	"go.viam.com/rdk/spatialmath"
 )
 
 // WorldState is a struct to store the data representation of the robot's environment.
 type WorldState struct {
-	Obstacles  []*GeometriesInFrame
-	Transforms []*LinkInFrame
+	names        map[string]bool
+	unnamedCount int
+	obstacles    []*GeometriesInFrame
+	Transforms   []*LinkInFrame
```

Comment:
I think if I am making everything else private this should also become private

Reply:
Considering the incoming GPS stuff, it might actually make sense to just make WorldState an interface here. We'll then have constructors for the "canonical" struct as you have here, plus a gps-based one that handles internal conversions to e.g. meters

Reply:
Interesting idea to make it an interface.  I think this gets us 90% of the way there by making things private.  I'm a little hesitant to actually define the interface until we need it though. Thoughts?

Reply:
Would be OK to defer that until after we get a better idea of what GPS Nav will entail. Otherwise, definitely agree on the private change.

Reply:
I'm fine with deferring on interface so long as this becomes private.